### ISSUE TITLE
add missing suggested exercises for #12in23 Mechanical March overachiever badge

### ISSUE THREAD

**Issue by DerTee** [exercism/zig#201]
TLDR: We should implement the exercises **Linked List** and **Sieve** to have the full set of suggested exercises for a #12in23 Mechanical March language.

## Background
Currently, for Mechanical March, which Zig is a part of (yay!!), there are five suggested exercises as per (source https://exercism.org/challenges/12in23):

- [ ]  Linked List*: Practice allocating/freeing memory, and pointers
- [X]  Secret Handshake: Explore bitwise operators
- [X]  Pangram: Explore for loops, strings and characters
- [ ]  Sieve: Practice arrays and for loops
- [X]  Binary Search: A classic exercise for loops.

Three of those already exist, but *Linked List* and *Sieve* are still missing, so you can currently not complete the #12in23 
 challenge "overachiever badge" for those people who complete all 5 suggested exercises for the respective language they chose for the themed months. In theory it would be sufficient to implement the missing exercises within the course of this year, but in practice it would be great to implement them within this month.
To avoid confusion: The "overachiever badge", which has no actual name yet, does currently not exist, it is just planned.

I'd be willing to implement both of the missing exercises, if they aren't already being worked on.
